TOOLS/lua/autocrop.lua

```diff
@@ -164,6 +182,15 @@ function detect_crop()
         return
     end
 
+    if options.use_vo_crop then
+        local hwdec_current = mp.get_property("hwdec-current")
+        if hwdec_current:find("-copy$") == nil and hwdec_current ~= "no" and
+           hwdec_current ~= "crystalhd" and hwdec_current ~= "rkmpp" then
```

Comment:
Could you explain why we need special case for `crystalhd` and `rkmpp`?

Reply:
I have no idea what either of those are used for, but according to the manpage `crystalhd` already copies video back to system RAM so there is no need to change hwdec and potentially switch to a different `hwdec-current`, same as for any other "-copy$" hwdec. I am not sure if it is the same for `rkmpp` since the manpage doesn't say if it copies to RAM, but even if it doesn't I suspect it is only for devices that have no alternative copy-back hwdec anyway.